pkg/storegateway/series_refs_streaming.go

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package storegateway
+
+type iteratorFactory func(strategy seriesIteratorStrategy) iterator[seriesChunkRefsSet]
```

Comment:
since we don't use different packages for chunks and labels this should be something like `seriesChunkRefsIteratorFactory` 🙁

Reply:
Good point, fixed in v